dotnet/samples/KernelSyntaxExamples/Example59_OpenAIFunctionCalling.cs

```diff
@@ -70,40 +68,48 @@ await foreach (var update in kernel.InvokePromptStreamingAsync("Given the curren
         WriteLine("======== Example 3: Use manual function calling with a non-streaming prompt ========");
         {
             var chat = kernel.GetRequiredService<IChatCompletionService>();
-            var chatHistory = new ChatHistory();
 
             OpenAIPromptExecutionSettings settings = new() { ToolCallBehavior = ToolCallBehavior.EnableKernelFunctions };
+
+            var chatHistory = new ChatHistory();
             chatHistory.AddUserMessage("Given the current time of day and weather, what is the likely color of the sky in Boston?");
+
             while (true)
             {
-                var result = (OpenAIChatMessageContent)await chat.GetChatMessageContentAsync(chatHistory, settings, kernel);
-
+                ChatMessageContent result = await chat.GetChatMessageContentAsync(chatHistory, settings, kernel);
                 if (result.Content is not null)
                 {
                     Write(result.Content);
                 }
 
-                List<ChatCompletionsFunctionToolCall> toolCalls = result.ToolCalls.OfType<ChatCompletionsFunctionToolCall>().ToList();
-                if (toolCalls.Count == 0)
+                IEnumerable<FunctionCallContent> functionCalls = result.Items.OfType<FunctionCallContent>(); // Getting list of function calls.
+                if (!functionCalls.Any())
                 {
                     break;
                 }
 
-                chatHistory.Add(result);
-                foreach (var toolCall in toolCalls)
+                chatHistory.Add(result); // Adding LLM response containing function calls(requests) to chat history as it's required by LLMs.
+
+                foreach (var functionCall in functionCalls)
                 {
-                    string content = kernel.Plugins.TryGetFunctionAndArguments(toolCall, out KernelFunction? function, out KernelArguments? arguments) ?
-                        JsonSerializer.Serialize((await function.InvokeAsync(kernel, arguments)).GetValue<object>()) :
-                        "Unable to find function. Please try again!";
-
-                    chatHistory.Add(new ChatMessageContent(
-                        AuthorRole.Tool,
-                        content,
-                        metadata: new Dictionary<string, object?>(1) { { OpenAIChatMessageContent.ToolIdProperty, toolCall.Id } }));
+                    try
+                    {
+                        FunctionResultContent functionResult = await functionCall.InvokeAsync(kernel); // Executing each function.
+
+                        chatHistory.AddMessage(AuthorRole.Tool, new ChatMessageContentItemCollection() { functionResult }); // Adding function result to chat history.
+                    }
+                    catch (Exception ex)
+                    {
+                        chatHistory.AddMessage(AuthorRole.Tool, new ChatMessageContentItemCollection() { new FunctionResultContent(functionCall, ex) });
```

Comment:
What do you think about this convenience method

`chatHistory.AddMessage(FunctionCallResult.CreateChatMessage(functionCallRequest, ex))`

Reply:
Yep, good idea - added.